app/presenters/pafs_core/validation_presenter.rb

```diff
@@ -310,5 +311,19 @@ def risks_error
                 "and the households benefiting.")
       false
     end
+
+    private
+
+    def funding_calculator_correct_version?
+      tfile = Tempfile.new(["funding_calculator", ".xlsx"])
+      tfile.write IO.read fetch_funding_calculator_for(project).path
+      xlsx = Roo::Spreadsheet.open tfile.path
+      sheet = xlsx.sheet(xlsx.sheets.grep(/PF Calculator/i).first)
+      calculator_version = Check.new(sheet).calculator_version
+
+      tfile.unlink
+
+      Check::VERSION_MAP.include? calculator_version
```

Comment:
To confirm, we just need to check that `Check::VERSION_MAP` has a particular key (e.g. "v8") and not the version_text, e.g. "Version 8" in the value? What has changed vs the previous version logic?

Reply:
If the calculator has the correct text value in one of the cells (B3 for v8 and B4 for v9), then the ```Check::VERSION_MAP``` will return a key to say what version of the calculator it is. If those cells do not contain valid text, then the calculator_version will be nil, so this will be false. I could change this to be ```calculator_version.nil?``` to be more explicit about what is being checked here, as we don't care what key is being returned, just that one is.